CsharpExtrasTest/Extensions/DictionaryTest.cs

```diff
@@ -63,6 +64,46 @@ public void GivenStringIntDictionaryWhenValuesMappedToStringThenKeysRemainUnchan
             Assert.AreEqual(mockDict.Keys, mappedDictionary.Keys);
         }
 
+        [Test]
+        [Category("Unit")]
+        public void GivenDictionaryWhenAddWithKeyDerivedFromValueCalledThenCorrectItemAddedToDictionary()
+        {
+            //Arrange
+            IDictionary<string, int> mockDict = MockStrIntDictionary();

```

Comment:
Is there a reason you using this Mock dictionary instead of a regular dictionary? Is this dictionary empty on creation?

Reply:
Not really, I just used the method already available to have a dictionary with some items. I can change it to be a new empty dictionary.

Reply:
I would use an empty dictionary then. Otherwise the mock dictionary might happen to contain the value you are adding and checking for, invalidating the test.